### PR TITLE
[16.0][FIX] stock_picking_restrict_cancel_printed: move cancellation

### PR DESCRIPTION
**Pull request by jbaudoux**
Prevent cancellation of stock move if picking is printed

cc @santostelmo @rousseldenis

### CODE REVIEW

**Review by santostelmo — APPROVED** [review]

**Review by cyrilmanuel — APPROVED** [review]

**Comment by jbaudoux** [comment]
cc @rousseldenis

**Comment by cyrilmanuel** [comment]
Hi @rousseldenis  is it good for you ?

**Review by rousseldenis — APPROVED** [review]

File: stock_picking_restrict_cancel_printed/models/stock_move.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, models
+from odoo.exceptions import UserError
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    def action_cancel(self):
```
Comment (rousseldenis) [inline stock_picking_restrict_cancel_printed/models/stock_move.py]:
```suggestion
    def _action_cancel(self):
```

File: stock_picking_restrict_cancel_printed/models/stock_move.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        for move in self:
+            if (
+                move.picking_id.printed
+                and move.picking_type_id.restrict_cancel_if_printed
+            ):
+                raise UserError(
+                    _("You cannot cancel a transfer that is already printed.")
+                )
+        return super().action_cancel()
```
Comment (rousseldenis) [inline stock_picking_restrict_cancel_printed/models/stock_move.py]:
```suggestion
        return super()._action_cancel()
```

**Review by rousseldenis — CHANGES_REQUESTED** [review]

**Comment by rousseldenis** [comment]
@jbaudoux  @cyrilmanuel Tests are failing

**Comment by jbaudoux** [comment]
> @jbaudoux @cyrilmanuel Tests are failing

Should be fixed with your above comment.
It's now fixed

**Comment by cyrilmanuel** [comment]
hii @rousseldenis is it good for you ?

**Review by rousseldenis — APPROVED** [review]

**Comment by rousseldenis** [comment]
@jbaudoux Could you check tests ?

**Comment by jbaudoux** [comment]
@rousseldenis I fix the tests (I added a move in the picking...). Can you re-have a quick look?

**Review by rousseldenis — APPROVED** [review]

**Comment by rousseldenis** [comment]
@jbaudoux ready ?

**Pull request merged**